src/components/appointments/shop/shopAppointments.tsx

```diff
@@ -0,0 +1,97 @@
+import { AppointmentSelectors } from "@redux/selectors/appointmentSelectors";
+import AppointmentTypes from "@redux/types/appointmentTypes";
+import styles from "@styles/pages/appointments/ShopAppointments.module.css";
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { AppointmentStatus, IAppointment } from "../../../types/appointment";
+import AppointmentCard from "./appointmentCard";
+
+interface IAppointmentsProps {
+  appointmentTab: AppointmentStatus;
+}
+
+const ShopAppointments = (props: IAppointmentsProps) => {
+  const dispatch = useDispatch();
+
+  const appointments = useSelector(AppointmentSelectors.getAppointments);
+
+  const { appointmentTab } = props;
+
+  const [appointmentsMap, setAppointmentsMap] = useState<{
+    [key: string]: Array<IAppointment>;
+  }>({});
+
+  useEffect(() => {
+    dispatch({ type: AppointmentTypes.READ_APPOINTMENTS });
```

Comment:
dispatch using exported action 
```suggestion
    dispatch(readAppointments());
```